pkg/kubelet/network/hostport/hostport_syncer.go

```diff
@@ -41,27 +41,44 @@ const (
 	kubeHostportChainPrefix string = "KUBE-HP-"
 )
 
-type HostportHandler interface {
-	OpenPodHostportsAndSync(newPod *ActivePod, natInterfaceName string, activePods []*ActivePod) error
-	SyncHostports(natInterfaceName string, activePods []*ActivePod) error
+// HostportSyncer takes a list of PodPortMappings and implements hostport all at once
+type HostportSyncer interface {
+	// SyncHostports gathers all hostports on node and setup iptables rules enable them. And finally clean up stale hostports
+	SyncHostports(natInterfaceName string, activePodPortMappings []*PodPortMapping) error
+	// OpenPodHostportsAndSync opens hostports for a new PodPortMapping, gathers all hostports on
+	// node, sets up iptables rules enable them. And finally clean up stale hostports.
+	// 'newPortMapping' must also be present in 'activePodPortMappings'.
+	OpenPodHostportsAndSync(newPortMapping *PodPortMapping, natInterfaceName string, activePodPortMappings []*PodPortMapping) error
+}
+
+// PortMapping represents a network port in a single container
+type PortMapping struct {
+	Name          string
+	HostPort      int32
+	ContainerPort int32
+	Protocol      v1.Protocol
+	HostIP        string
 }
 
-type ActivePod struct {
-	Pod *v1.Pod
-	IP  net.IP
+type PodPortMapping struct {
```

Comment:
Please add godoc since it's public, same for methods below like NewHostportSyncer